services/web3/getWeb3.js

```diff
@@ -50,6 +52,7 @@ async function loadWeb3CustomRpc(rpc = "http://localhost:8545") {
 
   const provider = new Web3.providers.HttpProvider(rpc);
   web3 = new Web3(provider);
+  getPermission();
```

Comment:
Could you simply have await ethereum.enable() here? If it throws, let it bubble up to browser, no need for rewriting the error and unsetting web3. Web3 is still useful either they give permission or not.